teuthology/run_tasks.py

```diff
@@ -81,13 +81,25 @@ def run_tasks(tasks, ctx):
         timer = Timer()
     stack = []
     try:
+        skip_shipping_utilities = False
+        if ctx.config.get('ship_utilities_early', False):
+            taskname = "ship_utilities"
+            log.info('Running task %s...', taskname)
+            timer.mark('%s enter' % taskname)
+            manager = run_one_task(taskname, ctx=ctx, config=None)
+            if hasattr(manager, '__enter__'):
+                stack.append((taskname, manager))
+                manager.__enter__()
+            skip_shipping_utilities = True
+
```

Comment:
That's dirty hack... because this is common algorithm for running tasks. If you want to inject some job into the stack please go to 'teuthology/run.py get_initial_tasks'.

Reply:
For sure the run_tasks.py should not be touched for this purpose.

Reply:
Another question what was wrong when it was added in the install tasks. Since this tasks is pretty specific to the install task.

Reply:
As described below, the current code has potential bug that if there have two `install:` tasks, they will ship the utilites but that not means they really need them. If both the two `install:` tasks really need the utilities, so when the second task is unwound in first place in reverse order it will remove the utilities, while the first task may still need it when being unwound.